wallet/src/de/schildbach/wallet/service/BlockchainServiceImpl.java

```diff
@@ -1072,20 +1072,18 @@ public void run() {
         });
     }
 
-    boolean updatingDashPayState = false;
     private void updateDashPayState() {
-        if (blockchainIdentityData != null && !updatingDashPayState) {
+        if (blockchainIdentityData != null) {
```

Comment:
I would like to replace dependence on BlockchainServiceImpl for the timer that triggers DashPay related updates from Platform.  When the blockchain service is shutdown, the timer is also stopped and no further updates occur.  What is the best place to start a timer in the app that can be shut down when the app is closed?  This timer would only be active if the user has a username.

Reply:
As I mentioned above, I think we should convert the PlatformRepo to a singleton. After that the periodic updates could be triggered by PlatformRepo itself. Also, I recommend the use of Handler instead of Timer, like this:
```
Handler handler = new Handler();

private Runnable runnableCode = new Runnable() {
    @Override
    public void run() {
      // Do something here on the main thread
      Log.d("Handlers", "Called on main thread");
      // Repeat this the same runnable code block again another 2 seconds
      // 'this' is referencing the Runnable object
      handler.postDelayed(this, timeInterval);
    }
};

// Start the initial runnable task by posting through the handler
handler.post(runnableCode);

// Removes pending code execution
handler.removeCallbacks(runnableCode);
```
Here is more info about Handlers vs Timers on Android: https://stackoverflow.com/questions/20330355/timertask-or-handler#:~:text=3%20Answers&text=Handler%20is%20better%20than%20TimerTask,%2C%20here%2C%20and%20here).

Reply:
Thanks.  I have implemented this except for the `handler.removeCallbacks` -- I wasn't sure where to put that so that it will be executed when the app is closed down.